injector/src/main/java/edu/ucr/cs/riple/injector/Helper.java

```diff
@@ -53,8 +53,62 @@
 /** A utility class. */
 public class Helper {
 
+  /** Utility class to match {@link CallableDeclaration} with their signatures in {@code String}. */
+  public static class SignatureMatcher {
+
+    /** Simple name of the callable. */
+    private final String callableName;
+    /** List of parameters detected from signature in string. */
+    private final List<String> parameterTypes;
+
+    /**
+     * Constructor to make a matcher instance.
+     *
+     * @param signature Signature to process.
+     */
+    public SignatureMatcher(String signature) {
+      this.callableName = extractCallableName(signature);
+      this.parameterTypes = extractParamTypesOfCallableInString(signature);
```

Comment:
Are `extractCallableName` and `extractParamTypesOfCallableInString` used outside this class? If not, let's make them private methods of `SignatureMatcher` and consider extracting this class as a top level class in its own file.

Reply:
No they can be internal. Moved the Signature Matcher class to a top level class. Please find it here: c33e8b35e3db38385d303aacf237a2d5b7282677